tests/components/local_calendar/test_diagnostics.py

```diff
@@ -0,0 +1,76 @@
+"""Tests for diagnostics platform of local calendar."""
+
+from freezegun import freeze_time
+
+from homeassistant.core import HomeAssistant
+
+from .conftest import TEST_ENTITY, ClientFixture
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+TEST_ICS = """BEGIN:VCALENDAR
+PRODID:-//github.com/allenporter/ical//4.5.0//EN
+VERSION:***
+BEGIN:VEVENT
+DTSTAMP:20230313T190500
+UID:***
+DTSTART:19970714T110000
+DTEND:19970714T220000
+SUMMARY:***
+CREATED:20230313T190500
+RRULE:FREQ=DAILY
+SEQUENCE:***
+END:VEVENT
+END:VCALENDAR"""
+
+
+@freeze_time("2023-03-13 12:05:00-07:00")
+async def test_empty_calendar(
+    hass: HomeAssistant,
+    setup_integration: None,
+    hass_client: ClientSessionGenerator,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test diagnostics against an empty calendar."""
+    data = await get_diagnostics_for_config_entry(hass, hass_client, config_entry)
+    assert data == {
+        "system_timezone": "tzlocal()",  # Will by python system timezone
+        "timezone": "America/Regina",  # HomeAssistant timezone set by fixture
+        "now": "2023-03-13T13:05:00-06:00",
+        "ics": "",
+    }
```

Comment:
Note: `diagnostics` is a great platform to get familiar with syrupy (snapshots / SnapshotAssertion).

```suggestion
    assert data == snapshot
```